include/kodlab_mjbots_sdk/pid_controller.h

```diff
@@ -0,0 +1,648 @@
+/**
+ * @file pid.h
+ * @author Chandravaran Kunjeti (kunjeti@seas.upenn.edu)
+ * @author Raghavesh Viswanath (vrag@seas.upenn.edu)
+ * @brief Class implementation of PID
+ * @date 5/18/2023
+ * @copyright 2023 The Trustees of the University of Pennsylvania.
+ * All rights reserved.
+ *
+ */
+
+#pragma once
+
+#include <math.h>
+#include <array>
+#include <vector>
+#include <algorithm>
+#include <Eigen/Dense>
+#include <tuple>
+
+namespace kodlab {
+
+/**
+ * @brief PIDController class that allows you to use PID control.
+ */
+
+template <typename Scalar, int N=1>
+class PIDController {
+
+public:
+  typedef Eigen::Matrix<Scalar,N,1> VectorNS;
+  
+  /**
+   * @brief Construct a vector Pid object
+   * @param p_gain Proportional Gain
+   * @param i_gain Derivative Gain
+   * @param d_gain Integral Gain
+   * @param time_step Time difference between 2 updates
+   * @param deadband Used to set range within setpoints where no correction 
+   * input is applied
+   * @param accumulator_limit Used to prevent runaway integrator/windup
+   * @param saturation controller output limits
+   */
+  PIDController(const VectorNS & p_gain,
+                const VectorNS & i_gain,
+                const VectorNS & d_gain, 
+                double time_step = 0.001,
+                VectorNS deadband = 0.05*VectorNS::Ones(), 
+                VectorNS accumulator_limit = 
+                    std::numeric_limits<Scalar>::max()*VectorNS::Ones(),
+                std::array<VectorNS, 2> saturation = 
+                    { -std::numeric_limits<Scalar>::max()*VectorNS::Ones(), 
+                       std::numeric_limits<Scalar>::max()*VectorNS::Ones() })
+    : 
+      deadband_(deadband),
+      accumulator_limit_(accumulator_limit),
+      saturation_(saturation),
+      time_step_(time_step){
+
+    set_gains(p_gain, i_gain, d_gain); 
```

Comment:
All these variables probably should be initialized in the initializer list.  The setter functions are not actually doing anything.  Thoughts @jdcaporale

Reply:
Agreed, if they can be in the initializer list they should be.